src/utils.py

```diff
@@ -132,100 +154,68 @@ def transcribe(audio_path: str,
     return output
 
 
-def translate(transcription) -> List[TranscriptionElement]:
-    # Initialize the translator
-    translator = Translator()
-
-    translated_elements = []
-
-    # Iterate over each transcription element and translate the text
-    for element in transcription:
-        try:
-            # Translate the text (assuming the original text is in Ukrainian 'uk')
-            translation = translator.translate(element.text.strip(), src='ru', dest='en')
-
-            # Create a new TranscriptionElement with the translated text
-            translated_element = TranscriptionElement(
-                time_start=element.time_start,
-                time_end=element.time_end,
-                text=translation.text
-            )
-
-            translated_elements.append(translated_element)
-            print(f"Translated segment [{element.time_start:.2f} --> {element.time_end:.2f}]: {translation.text}")
-
-        except Exception as e:
-            print(f"Error translating segment [{element.time_start:.2f} --> {element.time_end:.2f}]: {e}")
-            # In case of error, append the original untranslated element
-            translated_elements.append(element)
-
-    print("Translation completed.")
+def translate(transcription: List[TranscriptionElement]) -> List[TranscriptionElement]:
+    """ Translates each text element of transcription to English """
+    # TODO: Update to handle big chunks of text -> Iteratively translate the text. MAX: ~15k chars
+    full_text = " ".join([element.text.strip() for element in transcription])
+    full_text_translated = ts.translate_text(full_text, to_language="en")
+    # TODO: Check with longer videos if WhisperX gives sentence-by-sentence transcription and it relates to Translation
+    translated_sentences = nltk.tokenize.sent_tokenize(full_text_translated)
+    translated_elements = [
+        TranscriptionElement(
+            time_start=transcription_element.time_start,
+            time_end=transcription_element.time_end,
+            text=sentence
+        ) for sentence, transcription_element in zip(translated_sentences, transcription)
+    ]
     return translated_elements
 
 
+def create_voice_samples_dataset(audio_path: str,
+                                 translation: List[TranscriptionElement],
+                                 output_dir: str = "results/original_audio") -> str:
+    """ Creates small atomic audio files from big audio by `audio_path` based on time-stamps from translation """
```

Comment:
Change to:  Creates small atomic audio files from original audio file by `audio_path` based on time-stamps from translation